jetty-server/src/test/java/org/eclipse/jetty/server/ResponseTest.java

```diff
@@ -624,7 +635,7 @@ public void testResetContentTypeWithCharacterEncoding() throws Exception
 
         response.setContentType("wrong/answer;charset=utf-8");
         response.setContentType("foo/bar");
-        assertEquals("foo/bar", response.getContentType());
+        assertEquals("foo/bar;charset=utf-8", response.getContentType());
```

Comment:
Does this get sent over the wire on HTTP/1.1 as ...

```
Content-Type: foo/bar; charset=utf-8
```

?  

If so, then this PR is either doing too much, or we lack extra tests to allow a user to turn off that `charset=` portion on the HTTP response.

Reply:
This change won't force a change to the charset: in fact, the change *prevents* a change to the charset that that spec does not support.  Unfortunately, the javadoc for ServletResponse.setContentType() does not make this behaviour clear in the case of the content-type without char-set, so all we have to go on is the tck test: https://github.com/eclipse-ee4j/jakartaee-tck/blob/master/src/com/sun/ts/tests/servlet/api/common/response/ResponseTests.java#L680

Reply:
I created a new branch (from this one) called [`jetty-10.0.x-4719-keepcharencoding-extratests`](https://github.com/eclipse/jetty.project/tree/jetty-10.0.x-4719-keepcharencoding-extratests) with new testcases to show the concern (see commit e0b322f7d556837986accb3e881d5ec374285ede).

Two of the new testcases in ResponseHeadersTest are now failing ...

* testCharsetChangeToJsonMimeType()
* testCharsetChangeToJsonMimeTypeSetCharsetToNull()

> Interesting aside, I didn't know you could use `HttpServletResponse.reset()` after obtaining a `getWriter()` and still use it.

Reply:
@joakime I've merged your branch into mine, and added a fix for content types like json that must have an assumed char set. Thanks for the concrete tests!!